core/src/main/java/cz/xtf/core/bm/NamespaceManager.java

```diff
@@ -0,0 +1,140 @@
+    }
+
+    /**
+     * Used only in case when test cases are executed in different namespaces when:
+     * -Dxtf.openshift.namespace.per.testcase=true
+     *
+     * @return return namespace for currently running testcase or null if not present
+     */
+    public static String getNamespaceForTestcase(String testcase) {
+        return getTestcaseToNamespaceMap().get(testcase);
+    }
+
+    public static boolean createIfDoesNotExistsProject(String namespace) {
+        OpenShift openShift = OpenShifts.master(namespace);
+
+        boolean isCreated = false;
+        if (openShift.getProject() == null) {
+            log.info("Creating namespace: " + openShift.getNamespace());
+            openShift.createProjectRequest();
+            openShift.waiters().isProjectReady().waitFor();
+
+            try {
+                // Adding a label can be only done via 'namespace'. It cannot be set via 'project' API. Thus we do this
+                // separately. Also, to update namespace label, it's necessary to have 'patch resource "namespaces"'
+                // permission for current user and updated namespace, e.g. by having 'cluster-admin' role.
+                // Otherwise you can see:
+                // $ oc label namespace <name> "label1=foo"
+                // Error from server (Forbidden): namespaces "<name>" is forbidden: User "<user>" cannot patch resource "namespaces" in API group "" in the namespace "<name>"
+                OpenShifts.admin(namespace).namespaces().withName(openShift.getProject().getMetadata().getName())
+                        .edit(new Visitor<NamespaceBuilder>() {
+                            @Override
+                            public void visit(NamespaceBuilder builder) {
+                                builder.editMetadata()
+                                        .addToLabels(OpenShift.XTF_MANAGED_LABEL, "true");
+                            }
+                        });
+            } catch (KubernetesClientException e) {
+                // We weren't able to assign a label to the new project. Let's just print warning since this information
+                // is not critical to the tests execution. Possible cause for this are insufficient permissions since
+                // some projects using XTF are executed on OCP instances without 'admin' accounts available.
+                log.warn("Couldn't assign label '" + OpenShift.XTF_MANAGED_LABEL + "' to the new project '"
+                        + openShift.getNamespace() + "'. Possible cause are insufficient permissions.");
+                log.debug(e.getMessage());
+            }
+            isCreated = true;
+        }
+        if (OpenShiftConfig.pullSecret() != null) {
+            openShift.setupPullSecret(OpenShiftConfig.pullSecret());
+        }
+        log.info("Created namespace: " + openShift.getNamespace());
+        return isCreated;
+    }
+
+    public static boolean deleteProject(String namespace, boolean waitForDeletion) {
```

Comment:
Why can't be this method left in `ProjectCreator`?

Reply:
Same reason as above.